### PR TITLE
[ticket/13882] Lazy load the notification avatars.

### PR DESCRIPTION
**Pull request by callumacrae**
This moves them further down the waterfall, making the page load
quicker.

sorry for my crap php

https://tracker.phpbb.com/browse/PHPBB3-13882

### CODE REVIEW

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -4850,7 +4851,25 @@ function phpbb_get_avatar($row, $alt, $ignore_config = false)
...
+			$board_url = generate_board_url() . '/';
+			// This path is sent with the base template paths in the assign_vars()
+			// call below. We need to correct it in case we are accessing from a
+			// controller because the web paths will be incorrect otherwise.
+			$phpbb_path_helper = $phpbb_container->get('path_helper');
+			$corrected_path = $phpbb_path_helper->get_web_root_path();
+
+			$web_path = (defined('PHPBB_USE_BOARD_URL_PATH') && PHPBB_USE_BOARD_URL_PATH) ? $board_url : $corrected_path;
+
+			$theme = "{$web_path}styles/" . rawurlencode($user->style['style_path']) . '/theme';
```
Comment (callumacrae) [inline phpBB/includes/functions.php]: This is literally just `{T_THEME_PATH}` is there any easier way to do this?

Comment (nickvergessen) [inline phpBB/includes/functions.php]: Extract into a method/function somewhere?

Comment (callumacrae) [inline phpBB/includes/functions.php]: Most of it is reused, I think extracting it would add _more_ complexity

**Comment by marc1706** [comment]
Looks good. @callumacrae mind changing the last commit to fit our commit format?

**Comment by callumacrae** [comment]
doneee

File: phpBB/includes/functions_compatibility.php [hunk 2]
```diff
@@ -30,10 +30,11 @@
 * @param string $avatar_height Height of users avatar
 * @param string $alt Optional language string for alt tag within image, can be a language key or text
 * @param bool $ignore_config Ignores the config-setting, to be still able to view the avatar in the UCP
+* @param bool $lazy If true, will be lazy loaded (requires JS)
 *
 * @return string Avatar image
 */
-function get_user_avatar($avatar, $avatar_type, $avatar_width, $avatar_height, $alt = 'USER_AVATAR', $ignore_config = false)
+function get_user_avatar($avatar, $avatar_type, $avatar_width, $avatar_height, $alt = 'USER_AVATAR', $ignore_config = false, $lazy = false)
```
Comment (nickvergessen) [inline phpBB/includes/functions_compatibility.php]: this is a legacy function.
No need to add new features here?!

File: phpBB/assets/javascript/core.js [hunk 3]
```diff
@@ -1612,6 +1612,18 @@ phpbb.registerPageDropdowns = function() {
 	});
 };
```
Comment (marc1706) [inline phpBB/assets/javascript/core.js]: Add a tiny doc block here?

File: phpBB/phpbb/user_loader.php [hunk 4]
```diff
@@ -193,7 +194,7 @@ public function get_avatar($user_id, $query = false)
 			include($this->phpbb_root_path . 'includes/functions_display.' . $this->php_ext);
 		}
 
-		return get_user_avatar($user['user_avatar'], $user['user_avatar_type'], $user['user_avatar_width'], $user['user_avatar_height']);
+		return get_user_avatar($user['user_avatar'], $user['user_avatar_type'], $user['user_avatar_width'], $user['user_avatar_height'], 'USER_AVATAR', false, $lazy);
```
Comment (marc1706) [inline phpBB/phpbb/user_loader.php]: Please phpbb_get_avatar() here. NO idea why get_user_avatar() is being used as that is a legacy function ...

File: phpBB/includes/functions_compatibility.php (hunk 2 above)
Comment (marc1706) [inline phpBB/includes/functions_compatibility.php]: Yes, please don't use the legacy function anymore ...

Comment (callumacrae) [inline phpBB/includes/functions_compatibility.php]: It's being used, I'm not just changing functions for fun!

File: phpBB/phpbb/user_loader.php (hunk 4 above)
Comment (callumacrae) [inline phpBB/phpbb/user_loader.php]: How do I use it?

File: phpBB/includes/functions_compatibility.php (hunk 2 above)
Comment (nickvergessen) [inline phpBB/includes/functions_compatibility.php]: @marc1706 you wrote the function, mind changing it yourself?

File: phpBB/assets/javascript/core.js (hunk 3 above)
Comment (callumacrae) [inline phpBB/assets/javascript/core.js]: done

**Comment by callumacrae** [comment]
@marc1706 @nickvergessen So what's happening here? Am I changing the code, or is @marc1706?

**Comment by marc1706** [comment]
I personally don't have an issue with adding another parameter to the compatibility function. It doesn't break anything. I guess just create another ticket for the issue with using the legacy function in the user loader. Otherwise this should be fine.

**Comment by Nicofuma** [comment]
@marc1706 bump merge it if it's fine please

**Comment by nickvergessen** [comment]
Let's do this on 3.1.6

**Comment by callumacrae** [comment]
:confetti_ball:

**Pull request merged**